src/main/java/com/icthh/xm/ms/entity/config/KafkaMetricsSet.java

```diff
@@ -0,0 +1,55 @@
+package com.icthh.xm.ms.entity.config;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricSet;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.DescribeTopicsOptions;
+import org.apache.kafka.clients.admin.DescribeTopicsResult;
+import org.apache.kafka.clients.admin.TopicDescription;
+import org.springframework.kafka.core.KafkaAdmin;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import static java.util.Objects.nonNull;
+
+@Component
+@Slf4j
+@RequiredArgsConstructor
+public class KafkaMetricsSet implements MetricSet {
+
+    private final KafkaAdmin kafkaAdmin;
+    private final ApplicationProperties applicationProperties;
+
+    @Override
+    public Map<String, Metric> getMetrics() {
+        Map<String, Metric> metrics = new HashMap<>();
+        metrics.put("connection.success", (Gauge) this::connectionToKafkaTopicsIsSuccess);
+        return metrics;
+    }
+
+    public Boolean connectionToKafkaTopicsIsSuccess() {
```

Comment:
it there another way to check if connection to Kafka is success or not besides create connection with AdminClient? 
What his method return for next case:
1. start entity, connection to kafka is established
2. shutdown kafka, entity connection is broken
3. start kafka, entity still can not reconnect
4. check connectionToKafkaTopicsIsSuccess()

probably it will be TRUE but the entity's connection ot KAfka is broken.